### PR TITLE
21 testing code coverage and integration

### LINKED ISSUE #21: Testing/ Code Coverage and integration
Develop comprehensive unit tests to validate the functionality of individual components, functions, and classes. Ensure tests cover all critical logic and edge cases. Achieve a minimum code coverage threshold of 80%. This includes coverage for both tested and untested parts of the code to ensure robustness.

- [x] have CLI pull URLs from URLS.txt
- [x] have CLI output scores to JSON
- [x] run syntax checker (https://piazza.com/class/lzvpabcdwx83b0/post/52)
- [x] Have test coverage match file (https://piazza.com/redirect/s3?bucket=uploads&prefix=paste%2Flzvpcoco39x4ff%2F59b381bb5116cddd0493097c9ee0a7004273a7dd45c2e833467c5c8a54e32697%2FSampleCLICommandsAndStdout.md)

### PR DESCRIPTION
**Pull request by galaxybomb23**
PR to close #21 

Adds checker to CI.

### CODE REVIEW

**Review by RushilShaw — APPROVED** [review]

File: .gitIgnore
Comment (Coop8) [inline .gitIgnore:1]: .gitignore not .gitIgnore right?

**Review by Coop8 — COMMENTED** [review]
Looks good, good work

**Review by Coop8 — APPROVED** [review]

File: .gitIgnore
Comment (RushilShaw) [reply .gitIgnore:1]: Yes, although I cannot find a source to confirm this. .gitIgnore doesn't work on either Tom or my linux machine, however .gitignore does.

**Review by RushilShaw — APPROVED** [review]

File: .gitIgnore
Comment (galaxybomb23) [reply .gitIgnore:1]: changed filename, but when pushed it changes case back to .gitIgnore
image

**Pull request merged**